src/System.Collections.Specialized/src/System/Collections/Specialized/ListDictionary.cs

```diff
@@ -154,15 +154,15 @@ public ICollection Values
             }
         }
 
-        public void Add(object key, object value)
+        public void Add(object key, object? value)
         {
             if (key == null)
             {
                 throw new ArgumentNullException(nameof(key));
             }
             version++;
-            DictionaryNode last = null;
-            DictionaryNode node;
+            DictionaryNode? last = null;
+            DictionaryNode? node;
```

Comment:
Nit: You could move this declaration to the for loop `for (DictionaryNode? node = head)`

Reply:
If that is adhere with the statement: `Each PR should contain only changes related to the nullable annotations/attributes, no other changes.` 😄

Reply:
`no other changes` refers to code behavior, this is just a small minor clean-up as you're already touching this code, but it is fine to leave as is.